test/Makefile

```diff
@@ -129,7 +159,10 @@ build: download $(WASMS)
 
 $(WASMS): | $(OBJDIRS)
 $(OBJDIR)/%.wasm: $(OBJDIR)/%.wasm.o $(INFRA_WASM_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
+ifeq ($(WASI_SNAPSHOT), preview2)
+	$(WASM_TOOLS) component new --adapt $(ADAPTER) $@ -o $@
```

Comment:
This is not very good practice. If `wasm-tools` fails here (for whatever reason), the Wasm file will not be changed, but it will be considered up to date for `make`, so this rule will not run on the next invocation to `make`. Ideally, you’d have something like a temporary intermediate file, so that the input and output to `wasm-tools` is a different file.

Something like this:

~~~
$(CC) $(CFLAGS) $(LDFLAGS) -o tmp.wasm $^
$(WASM_TOOLS) component new --adapt $(ADAPTER) tmp.wasm -o $@
$(RM) tmp.wasm
~~~